compiler/rustc_resolve/src/late/lifetimes.rs

```diff
@@ -482,6 +482,11 @@ fn convert_named_region_map(tcx: TyCtxt<'_>, named_region_map: NamedRegionMap) -
         let def_id = tcx.hir().local_def_id(hir_id);
         map.insert(def_id);
     }
+    for (_, late_bound) in &mut rl.late_bound {
+        late_bound.sort_by(|&a, &b| {
```

Comment:
nit: I think it would be good to leave a comment here explaining why we're doing this sort